pkgs/development/compilers/llvm/7/compiler-rt.nix

```diff
@@ -55,6 +57,11 @@ stdenv.mkDerivation rec {
     ln -s $out/lib/*/cclang_rt.crtend-*.o $out/lib/linux/crtend.o
     ln -s $out/lib/*/clang_rt.crtbegin_shared-*.o $out/lib/linux/crtbeginS.o
     ln -s $out/lib/*/clang_rt.crtend_shared-*.o $out/lib/linux/crtendS.o
+  '' + stdenv.lib.optionalString stdenv.hostPlatform.isWasm ''
+    for f in crtbegin crtend crtbeginS crtendS; do
+      touch $f.c
+      $CC -c -o $out/lib/$f.o $f.c
+    done
```

Comment:
This was a horrible hack when I did it :P Any ideas on making this not horrible?

Reply:
Originally I thought we could just pass ```-nostartfiles``` everywhere but I think there might be some other crt files it needs?

Reply:
Actually doesn't seem to be needed! recent clang must just be skipping crtbegin and crtend for webassembly!